src/vs/workbench/browser/parts/views/viewsViewlet.ts

```diff
@@ -35,7 +35,17 @@ import { Registry } from 'vs/platform/registry/common/platform';
 
 export abstract class TreeViewsViewletPanel extends ViewletPanel {
 
-	protected tree: WorkbenchTree;
+	private _tree: WorkbenchTree;
+	protected get tree(): WorkbenchTree { return this._tree; }
+	protected set tree(tree: WorkbenchTree) {
+		this._tree = tree;
+		this.disposables.push(this._tree);
+	}
+
+	setInput(element): TPromise<any, any> {
```

Comment:
This is bit weird that setInput has the behaviour of focussing the first element. What happens if the input is empty and elements come later? 

It might be clean to have a method `initializeFocus` or `resetFocus` which is called when

- tree setter and if tree has elements
- When the elements are added
- When the focus is set to empty by removing the existing focussed element

Also please consider [CustomTreeViews](https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/browser/parts/views/customView.ts#L39).

Reply:
@sandy081 I attempted this with list views over in #53616, but found it messy to implement as there is no generalized way to get change notifications from `Lists`s or `Tree`s that I know of. Do you have suggestions for doing this in a generalizable way?

Reply:
@JacksonKearl @joaomoreno Why not add this feature (focus first element by default) in Tree widget as a behaviour to opt in ?